### PR TITLE
Remove DAGNode deprecations

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Removes DAGNode deprecations

### Details and comments

Since 0.20.0 has been released and the 3 months have passed, this PR removes the deprecations to DAGNode that were added as part of #6567.

### CODE REVIEW

**Review by kdk — COMMENTED** [review]
Thanks @enavarro51 , it looks like there's a section in the tutorial https://github.com/Qiskit/qiskit-tutorials/blob/721d67ac97e8f0ca3b9b4156cc35a9a777363d12/tutorials/circuits_advanced/04_transpiler_passes_and_passmanager.ipynb that still uses the deprecated `node.condition` form and is leading to the failure at https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=35731&view=logs&j=b32a9d47-7bd3-5443-5e7c-fab46a446055&t=186ae8ed-dea6-5bc5-f4e3-12f54632f4a9&l=60 . That would need to be updated before this PR can merge.

**Review by kdk — COMMENTED** [review]
Thanks @enavarro51 , this looks good to me. Can you add a "removal" release note?

**Comment by enavarro51** [comment]
Actually as I looked at this, should we be leaving the `name` getter and setter? I thought we were going to still allow `node.name`. Thanks.

**Comment by enavarro51** [comment]
@kdk Ok. Release note done and I put the name setter and getter back.

**Comment by enavarro51** [comment]
@kdk. NVM. Just realized name getter and setter were now in DAGOpNode. Should be all good now.

File: releasenotes/notes/remove-dagnode-deprecations-30703a2156d52b8a.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+upgrade:
+  - |
+    For the :class:`~DagNode`, the previously deprecated ``type``, ``op``,
+    ``qargs``, ``cargs``, and ``wire`` kwargs and attributes have been removed.
+    These classes were originally deprecated in the 0.19.0 release. The ``op``,
+    ``qargs``, and ``cargs`` kwargs and attributes can be accessed through the
+    :class:`~DagOpNode`, and the ``wire`` kwarg and attribute can be accessed
+    through the :class:`DagInNode` or :class:`DagOutNode`.
```
Comment (kdk) [inline releasenotes/notes/remove-dagnode-deprecations-30703a2156d52b8a.yaml:9]:
```suggestion
    through the :class:`~DagInNode` or :class:`~DagOutNode`.
```

**Review by kdk — APPROVED** [review]
Thanks @enavarro51 , this looks good to me!

**Review by kdk — APPROVED** [review]

**Comment by jakelishman** [comment]
There was a seemingly bad-luck test failure on this on Windows/Python3.7, so I've restarted that, and it should go through now.

**Pull request merged**